include/seastar/http/exception.hh

```diff
@@ -65,10 +65,21 @@ private:
  */
 class redirect_exception : public base_exception {
 public:
-    redirect_exception(const std::string& url, http::reply::status_type status = http::reply::status_type::moved_permanently)
-            : base_exception("", status), url(url) {
+    redirect_exception(const std::string& url, http::reply::status_type status = http::reply::status_type::moved_permanently, const std::optional<int>& retry_after = std::nullopt)
+            : base_exception("", status), url(url), retry_after(retry_after) {
```

Comment:
You should mention in the commit message (and perhaps even split the commit into two) that one of the things that this patch does is to add to add Retry-After header (see https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.3) support for redirections.
It is separate feature from the to_reply() function you are adding in this patch.